text/0000-compiler-team-rolling-leads.md

```diff
@@ -0,0 +1,282 @@
+
+In code:
+
+```
+enum Level { Senior, Junior }
+struct Member { lead: Option<Level>, ... }
+
+fn roll(curr_senior: &mut Member, curr_junior: &mut Member, incoming: &mut Member) {
+    assert_eq!(curr_senior.lead, Some(Level::Senior));
+    assert_eq!(curr_junior.lead, Some(Level::Junior));
+    assert_ne!(curr_senior, incoming);
+    assert_ne!(curr_junior, incoming);
+    curr_senior.lead = None;
+    curr_junior.lead = Some(Level::Senior);
+    incoming.lead = Some(Level::Junior);
+}
+```
+
+### Selection process
+
+When the senior co-lead decides that they are ready to step down, and have
+confirmed that the junior co-lead feels ready to take on the senior co-lead
+role, then the two tell the T-compiler team privately about the intention to
+have a rollover in leadership.
+
+Then the T-compiler team members can nominate their teammates to serve as the
+new junior co-lead. We here follow the model of our FCP process: The leads
+should provide a ten-day window for nominations to come in, unless they get
+confirmation that the set of nominations is complete.
+
+After nomination is completed, the outgoing senior and junior co-leads discuss
+the set of nominees, and also, if desired, have short discussions with the
+nominees. Then, the senior and junior co-leads select the new co-lead from the
+set of nominees. And that’s it! Then current leads just need to publish a blog
+post saying that the leadership is scheduled to roll over, who the new junior
+lead is, and the date that it takes effect.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+Committing to specific term lengths puts pressure on the leads to identify new
+leaders earlier than they might otherwise.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Rolling, not Rotation
+
+We specify here a rolling leadership process, where we will see a shift in
+leads, but not every team member is required to serve as a leader at any point.
+An alternative model is a true *rotating leadership*, where every team member
+will eventually become a lead.
+
+Our rationale for this is that we do not think every team member is interested
+in becoming a lead. If we adopted a true rotation and forced someone to become a
+lead who did not really want the role, then that probably be bad for that
+individual, and probably would be bad for the team as well.
+
+## Selection, Not Election
+
+We specify here that the new junior lead is *selected* (from a set of
+individuals nominated by the compiler team) by the current leads. An obvious
+alternative would be a pure democracy where the electorate (either the compiler
+team, or some superset thereof) gets to vote for who the new junior lead will
+be.
+
+When it comes to co-leadership, the two leaders need to be able to work
+together effectively. We do not want to end up with a pair of leaders who cannot
+talk to each other. Therefore, we currently are choosing a system where the
```

Comment:
> a pair of leaders who cannot talk to each other

The point stands, but having two compiler team members not able to talk to each other is already a red flag. The working relationship of the two co-leads is already a consideration even if they are capable of talking to each other 😉. Maybe it would be a good idea to soften the language here to indicate that the skills and working styles of the two co-leads need to fit well together.

Reply:
Yes, you're right; I used imperfect language here and will soften it.

Reply:
Hopefully the text added in 23bf9947662b306a1f4675b8e0529ed42d779a8a addresses the point raised here.